pkg/scheduler/cache/event_handlers.go

```diff
@@ -66,6 +73,128 @@ func (sc *SchedulerCache) getOrCreateJob(pi *schedulingapi.TaskInfo) *scheduling
 	return sc.Jobs[pi.Job]
 }
 
+func (sc *SchedulerCache) addPodCSIVolumesToTask(pi *schedulingapi.TaskInfo) error {
+	volumes, err := sc.getPodCSIVolumes(pi.Pod)
+	if err != nil {
+		klog.Errorf("got pod csi attachment persistent volumes count error: %s", err.Error())
+		return err
+	}
+
+	for key, count := range volumes {
+		pi.Resreq.AddScalar(key, float64(count))
+	}
+	return nil
+}
+
+func (sc *SchedulerCache) getPodCSIVolumes(pod *v1.Pod) (map[v1.ResourceName]int64, error) {
+	volumes := make(map[v1.ResourceName]int64)
+	for _, vol := range pod.Spec.Volumes {
+		pvcName := ""
+		isEphemeral := false
+		switch {
+		case vol.PersistentVolumeClaim != nil:
+			// Normal CSI volume can only be used through PVC
+			pvcName = vol.PersistentVolumeClaim.ClaimName
+		case vol.Ephemeral != nil:
+			// Generic ephemeral inline volumes also use a PVC,
+			// just with a computed name and certain ownership.
+			// That is checked below once the pvc object is
+			// retrieved.
+			pvcName = ephemeral.VolumeClaimName(pod, &vol)
+			isEphemeral = true
+		default:
+			// TODO: support Inline volumes.
```

Comment:
Please mark the submitter information in todo. Such as: "@Lily922 TODO: support Inline volumes."

Reply:
Done